vulnerabilities/importers/nvd.py

```diff
@@ -20,150 +20,175 @@
-        return max(summaries, key=len)
-
-    @staticmethod
-    def extract_severity_scores(cve_item):
-        severity_scores = []
-
-        if cve_item["impact"].get("baseMetricV3"):
-            severity_scores.append(
-                VulnerabilitySeverity(
-                    system=scoring_systems["cvssv3"],
-                    value=str(cve_item["impact"]["baseMetricV3"]["cvssV3"]["baseScore"]),
+                    reference_id=cpe,
                 )
             )
-            severity_scores.append(
-                VulnerabilitySeverity(
-                    system=scoring_systems["cvssv3_vector"],
-                    value=str(cve_item["impact"]["baseMetricV3"]["cvssV3"]["vectorString"]),
-                )
-            )
-
-        if cve_item["impact"].get("baseMetricV2"):
-            severity_scores.append(
-                VulnerabilitySeverity(
-                    system=scoring_systems["cvssv2"],
-                    value=str(cve_item["impact"]["baseMetricV2"]["cvssV2"]["baseScore"]),
-                )
-            )
-            severity_scores.append(
-                VulnerabilitySeverity(
-                    system=scoring_systems["cvssv2_vector"],
-                    value=str(cve_item["impact"]["baseMetricV2"]["cvssV2"]["vectorString"]),
-                )
+        references.append(
+            Reference(
+                url=f"https://nvd.nist.gov/vuln/detail/{cve_id}",
+                reference_id=cve_id,
+                severities=severity_scores,
             )
-
-        return severity_scores
-
-    def extract_reference_urls(self, cve_item):
-        urls = set()
-        for reference in cve_item["cve"]["references"]["reference_data"]:
-            ref_url = reference["url"]
-
-            if not ref_url:
-                continue
-
-            if ref_url.startswith("http") or ref_url.startswith("ftp"):
-                urls.add(ref_url)
-
-        return urls
-
-    def is_outdated(self, cve_item):
-        cve_last_modified_date = cve_item["lastModifiedDate"]
-        cve_last_modified_date_obj = dateparser.parse(cve_last_modified_date)
-
-        if self.config.cutoff_date:
-            return cve_last_modified_date_obj < self.config.cutoff_date
-
-        if self.config.last_run_date:
-            return cve_last_modified_date_obj < self.config.last_run_date
-
-        return False
-
-    def related_to_hardware(self, cve_item):
-        for cpe in self.extract_cpes(cve_item):
-            cpe_comps = cpe.split(":")
-            # CPE follow the format cpe:cpe_version:product_type:vendor:product
-            if cpe_comps[2] == "h":
-                return True
-
-        return False
-
-    @staticmethod
-    def extract_cpes(cve_item):
-        cpes = set()
-        for node in cve_item["configurations"]["nodes"]:
-            for cpe_data in node.get("cpe_match", []):
+        )
+        if "https://nvd.nist.gov/vuln/detail/{cve_id}" in ref_urls:
```

Comment:
Why do you remove the CVEs from references?
We still want them there IMHO ... in particular that's where we would get the severity score from the NVD?

Reply:
I will add a comment and a test in the code to explain this

Reply:
https://github.com/nexB/vulnerablecode/issues/679